coversation.py

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env python2
+
+import time
+from naoqi import ALProxy
+
+
+def repeat_test():
+    '''
+    test code - pepper repeats word back to user
+    '''
+    IP = "whatever the IP is"
+    tts = ALProxy("ALTextToSpeech", IP, 9559)  # initiates text to speech functionality
+    asr = ALProxy("ALSpeechRecognition", IP, 9559)  # initiates speech recognition functionality
+    memory = ALProxy("ALMemory")
+
+    asr.setlanguage("English")
+    vocab = ["good", "bad"]
+    asr.setVocabulary(vocab, False)  # sets what pepper understands
+
+    # start the speech recognition engine with user Test_ASR
+    # FIXME: not sure if this 'user' should be the username in the NAOqi system
+    asr.subscribe("Test_ASR")  # pepper listens
+
+    # get data
+    # FIXME: words understood go in WordRecognized key stored in ALMemory - how to access?
+    # retrieve list of understood phrases from ALMemory (maybe???)
+    heard = memory.getData("WordRecognized")

```

Comment:
Hi malik, I have reorganized your code a little bit. And I think here is some problem. I did not find a memory key called `WordRecognized` in the [Memory Key Index](https://developer.softbankrobotics.com/pepper-naoqi-25/naoqi-developer-guide/former-naoqi-framework/naoqi-memory-key-index), instead, I found [it is an event](https://developer.softbankrobotics.com/pepper-naoqi-25/naoqi-developer-guide/naoqi-apis/naoqi-audio/alspeechrecognition-0#-events--sbr-fake-anchor). So I think we should treat it as an event.

I collected some docs about events [here](https://github.com/RTCS-Team25/Project/tree/Development#event), and you can see that we are provided two methods to subscribe to the events. I suppose it could be much easier **with qi**.

Could you please change this part?